projects/ngx-universign/src/lib/model/universign.builder.ts

```diff
@@ -0,0 +1,89 @@
+import {RedirectionMode, UNIVERSIGN_DEFAULT_IFRAME_ID, UniversignEvent} from './universign.typings';
+import {Universign} from './universign.model';
+import {EventEmitter} from '@angular/core';
+
+/**
+ * Universign class builder
+ */
+export class UniversignBuilder {
+  private $el: string; // HTML identifier in which the iframe will be added
+  private $redirectionMode: RedirectionMode;
+  private $signerId: string; // Signer identifier (mandatory)
+  private $scriptUrl: string; // Iframe javascript url
+
+  // Universign events
+  private $onEnd: EventEmitter<UniversignEvent>;
+  private $onStart: EventEmitter<UniversignEvent>;
+  private $onChange: EventEmitter<UniversignEvent>;
+
+  static getInstance(): UniversignBuilder {
+    return new UniversignBuilder();
+  }
+
+  get el(): string {
+    return this.$el;
+  }
+
+  get redirectionMode(): RedirectionMode {
+    return this.$redirectionMode;
+  }
+
+  get signerId(): string {
+    return this.$signerId;
+  }
+
+  get scriptUrl(): string {
+    return this.$scriptUrl;
+  }
+
+  get end(): EventEmitter<UniversignEvent> {
+    return this.$onEnd;
+  }
+
+  get start(): EventEmitter<UniversignEvent> {
+    return this.$onStart;
+  }
+
+  get change(): EventEmitter<UniversignEvent> {
+    return this.$onChange;
+  }
+
+  setEl(el: string) {
```

Comment:
Pourquoi ne pas utiliser la syntaxe "setter" ?

```
set el(el: string) {
    this.$el = el || UNIVERSIGN_DEFAULT_IFRAME_ID;
    return this;
}
```

(valable aussi pour les autres setters

Reply:
car un setter n'est pas une fonction en ES6, dans ton cas je vais avoir:
`const builder = new UniversignBuilder().el = 'iframeContainer';`

De plus un setter ne peut pas retourner de valeur, je ne peux donc pas faire un return this pour rester dans la logique du builder

Reply:
ok, compris, il faudrait changer le prefix du coup, ce n'est pas un setter.

Reply:
done